icu4c/source/i18n/number_longnames.cpp

```diff
@@ -1410,13 +1405,15 @@ void MixedUnitLongNameHandler::forMeasureUnit(const Locale &loc,
                                               const MicroPropsGenerator *parent,
                                               MixedUnitLongNameHandler *fillIn,
                                               UErrorCode &status) {
+    U_ASSERT(mixedUnit.getComplexity(status) == UMEASURE_UNIT_MIXED);
     U_ASSERT(fillIn != nullptr);
     if (U_FAILURE(status)) {
         return;
     }
 
     MeasureUnitImpl temp;
     const MeasureUnitImpl &impl = MeasureUnitImpl::forMeasureUnit(mixedUnit, temp, status);
+    // Defensive, for production code:
     if (impl.complexity != UMEASURE_UNIT_MIXED) {
```

Comment:
Since we don't get read-only access to MeasureUnitImpl in ICU4J, we can't do this quick/cheap check for complexity in production code.